sys/aarch64/evec.S

```diff
@@ -165,6 +165,9 @@ ENTRY(handle_user_trap)
         mov     x0, sp
         bl      user_trap_handler
 user_exc_leave:
+        /* disable interrupts */
+        msr     daifset, (DAIF_I_MASKED >> 6)
```

Comment:
Well... this is confusing.

Some observations:
 * `msr daifset, #x` and `msr daifclr, #x` use `DAIF_I_MASKED` shifted right by `6`,
 * `mrs %r, daif` mirrors interrupt flags from current `PSR`,
 *  in our code `DAIF_I_MASKED` is the same as `PSR_I`.

Maybe we should clearly separate use of `PSR_*` and `DAIF_*` ? I would much prefer that `DAIF_I` (without `MASKED` suffix if possible) was used only with `daifclr` and `daifset` registers and `PSR_I` with `daif`. I'd add a comment in `armreg.h` about this oddity.

Reply:
done